### PR TITLE
Add spring-cloud extension

### PR DESCRIPTION
**Pull request by peizhou298**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [x] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
@zikalino can help with review.

File: src/spring-cloud/setup.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    'Programming Language :: Python :: 2.7',
+    'Programming Language :: Python :: 3',
+    'Programming Language :: Python :: 3.4',
+    'Programming Language :: Python :: 3.5',
+    'Programming Language :: Python :: 3.6',
+    'License :: OSI Approved :: MIT License',
+]
+
+# TODO: Add any additional SDK dependencies here
+DEPENDENCIES = [
```
Comment (zikalino) [inline src/spring-cloud/setup.py]: just leave it empty [] otherwise it will cause problem with release

Comment (peizhou298) [reply src/spring-cloud/setup.py]: Fixed alrady, remove 'azure-cli-core' int dependencies.

File: src/front-door/azext_front_door/custom.py [hunk 2]
```diff
@@ -861,11 +861,11 @@ def add_custom_rule_match_condition(cmd, resource_group_name, policy_name, rule_
                 selector = variable_parts[1]
 
             rule.match_conditions.append(MatchCondition(
-                match_variable=match_variable,
-                selector=selector,
```
Comment (zikalino) [inline src/front-door/azext_front_door/custom.py]: I know this is only whitespace, but could we just somehow revert it?

**Comment by zikalino** [comment]
are you planning to replace SDK with newest version?

File: src/front-door/azext_front_door/custom.py (hunk 2 above)
Comment (peizhou298) [reply src/front-door/azext_front_door/custom.py]: Sure, wired thing is I copy the source text from main branch but still get this problem, let me double fix.

**Comment by peizhou298** [comment]
> are you planning to replace SDK with newest version?

**Comment by peizhou298** [comment]
Reopen.

**Pull request merged**